### PR TITLE
Add Konbini to PaymentSheet + PI

### PR DESCRIPTION
**Pull request by yuki-stripe**
## Summary

https://github.com/stripe/stripe-ios/assets/47796191/a83ac20a-06f0-43bf-ac91-ed0ea9006902

Note both Checkout and web PE don't use their normal 'phone number' field to collect it - it's just a normal text field (11 digits max, numbers only) and doesn't interact with merchant configuration related to the collection of a phone number (e.g. Checkout shows a separate phone field if you tell it to collect a phone number).

## Motivation
https://docs.google.com/spreadsheets/d/1-cSawAjjWBsJ0XawY7qj-NOeMcdjsdd8ur_MyBz5DBU/edit#gid=0 

## Testing
- Konbini confirm flow tests
- Manually tested PaymentSheet and PaymentSheet.FlowController

Full billing details collection:
CleanShot 2023-09-13 at 10 23 49@2x

CleanShot 2023-09-13 at 10 24 23@2x
CleanShot 2023-09-13 at 10 56 13@2x

CleanShot 2023-09-13 at 10 56 33@2x

### CODE REVIEW

File: StripePayments/StripePayments/Source/API Bindings/Models/Shared/STPIntentActionKonbiniDisplayDetails.swift [hunk 1]
```diff
@@ -0,0 +1,45 @@
+//
+//  STPIntentActionKonbiniDisplayDetails.swift
+//  StripePayments
+//
+//  Created by Yuki Tokuhiro on 9/12/23.
+//
+
+import Foundation
+
+/// Contains OXXO details necessary for the customer to complete the payment.
```
Comment (jaynewstrom-stripe) [inline StripePayments/StripePayments/Source/API Bindings/Models/Shared/STPIntentActionKonbiniDisplayDetails.swift:10]:
```suggestion
/// Contains Konbini details necessary for the customer to complete the payment.
```

File: StripePayments/StripePayments/Source/API Bindings/Models/PaymentMethods/STPPaymentMethodEnums.swift [hunk 2]
```diff
@@ -151,6 +153,8 @@ import Foundation
             return "Amazon Pay"
         case .mobilePay:
             return "MobilePay"
+        case .konbini:
+            return STPLocalizedString("Konbini", "Payment Method type brand name")
```
Comment (porter-stripe) [inline StripePayments/StripePayments/Source/API Bindings/Models/PaymentMethods/STPPaymentMethodEnums.swift:157]: Why don't we have a localized string for the other LPMs we converted from .dynamic? But added here?

File: StripePayments/StripePayments/Source/API Bindings/Models/Shared/STPIntentActionKonbiniDisplayDetails.swift [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+//
+//  Created by Yuki Tokuhiro on 9/12/23.
+//
+
+import Foundation
+
+/// Contains Konbini details necessary for the customer to complete the payment.
+public class STPIntentActionKonbiniDisplayDetails: NSObject, STPAPIResponseDecodable {
+    /// The date at which the pending Konbini payment expires.
+    @objc public let expiresAt: Date
```
Comment (porter-stripe) [inline StripePayments/StripePayments/Source/API Bindings/Models/Shared/STPIntentActionKonbiniDisplayDetails.swift:13]: Why do we parse this? Is it used somewhere?

**Review by porter-stripe — APPROVED** [review]

File: StripePayments/StripePayments/Source/API Bindings/Models/Shared/STPIntentActionKonbiniDisplayDetails.swift (hunk 3 above)
Comment (yuki-stripe) [reply StripePayments/StripePayments/Source/API Bindings/Models/Shared/STPIntentActionKonbiniDisplayDetails.swift:13]: It's not used but it's in the public API.

File: StripePayments/StripePayments/Source/API Bindings/Models/PaymentMethods/STPPaymentMethodEnums.swift (hunk 2 above)
Comment (yuki-stripe) [reply StripePayments/StripePayments/Source/API Bindings/Models/PaymentMethods/STPPaymentMethodEnums.swift:157]: The previously .dynamic ones didn't use localized strings. Konbini is written differently in Japanese, though.

**Pull request merged**